src/BookReader/utils.js

```diff
@@ -40,6 +40,23 @@ export function notInArray(value, array) {
   return !array.includes(value);
 }
 
+/**
+ * Determines the active element, going into shadow doms.
+ * @return {Element}
+ */
+export function getActiveElement(doc = document, recurseShadowDom = true) {
+  const activeElement = doc.activeElement;
+  if (recurseShadowDom && activeElement?.shadowRoot) {
+    return getActiveElement(activeElement.shadowRoot, true);
```

Comment:
smol q: this shouldn't be too expensive, ya?

Reply:
Hopefully? :P It should only have to go through I'd guess like max 10? We might want to optimize to only run this if the key pressed is on of interest. Right now it runs for any kepress, before determining if it's of interest. Let me see how hard that would be...

Reply:
maybe if we debounce/throttle that keypress event handler, it _may_ be enough..

Reply:
Is there a way to be more explicit about this than traversing the DOM? ie add a method to the components that can be active and call `component.isActive()`?

Reply:
@iisa Switched order of calls :+1:

We'd need something like `isKeyboardSink()` or something. This isn't the _best_ solution as is; I think there are ways one can fully block shadow DOM access to your web component. But it solves the current problem for our components, and we can always change it later :) This also has the benefit of working for other people who might embed BR in their sites, which might not be using our web component methods.